django/library/serializers.py

```diff
@@ -75,13 +75,12 @@ def get_existing_contributor(self, validated_data):
                 Contributor.objects.filter(user__username=username).first(),
             )
         else:
-            # filter by given_name and family_name if they were entered
-            name_filter = {
-                k: v
-                for k, v in validated_data
-                if k in ("family_name", "given_name") and v
+            # find an existing contributor that exactly matches the given name and email
+            contrib_filter = {
+                k: validated_data.get(k, "")
+                for k in ["given_name", "family_name", "email"]
```

Comment:
the dict comprehension was to create filter criteria like

 `{"email": "foo@example.com"}`

instead of 

 `{ "given_name": "", "family_name": "", "email": "foo@example.com" }`

wouldn't the latter prevent a record that matches the email but also has a given name and family name from matching?

Reply:
Oh right, we should always match emails, I was just thinking about name conflicts

maybe
```
if email:
  get by email
else:
  filter by name + blank email (?)
```